### PR TITLE
added ability to pass multiple dirs to the --path option

### LINKED ISSUE #37: Add ability to pass multiple directories to the --path option
The `--path` option currently only supports a single argument. This means if you have two sibling directories of tests, you'd have to specify a path of `.` in order for them to be found. This is really inefficient, because it means ward will spend a lot of time looking in places for tests where they won't be found.

A better interface would be e.g.

```
ward --path ./test_dir_one ./test_dir_two
```

Command line input in this project is handled by the click library, which supports this functionality out of the box: https://click.palletsprojects.com/en/7.x/options/#multi-value-options

To implement this change, one option would be to alter the `get_info_for_modules` to take a `List[str]` instead of a `str` argument:

https://github.com/darrenburns/ward/blob/e323f7a3b08b1175694055d75c0554321b24f1e9/ward/collect.py#L16

Then passing that list through to `pkgutil.iter_modules`:

https://github.com/darrenburns/ward/blob/e323f7a3b08b1175694055d75c0554321b24f1e9/ward/collect.py#L18

### PR DESCRIPTION
**Pull request by AlecJ**
resolves #37 

Updated the click argument for run to accept any number of directories. I noticed while testing that it is possible to supply 0 arguments which may not be desired. Unfortunately, the click library does not have a simple way of accepting a variable number of args and so I had to use a workaround (see run.py).

Updated collect.py to handle a list of directories as well.

### CODE REVIEW

**Comment by darrenburns** [comment]
@AlecJ Thanks so much for the PR :) I can only have brief look today, but will review fully it asap!

File: ward/collect.py [hunk 1]
```diff
@@ -13,18 +13,22 @@ def is_test_module(module: pkgutil.ModuleInfo) -> bool:
     return module.name.startswith("test_")
 
 
-def get_info_for_modules(path: str) -> Generator[pkgutil.ModuleInfo, None, None]:
-    # Check for modules at the root of the specified path
-    for module in pkgutil.iter_modules([path]):
+def get_info_for_modules(path: List[str]) -> Generator[pkgutil.ModuleInfo, None, None]:
+    # If multiple paths are specified, remove duplicates
+    paths = list(set(path))
```
Comment (darrenburns) [inline ward/collect.py]: This raises an interesting question: what happens if you have two paths that aren't strictly equal, but one is contained within the other? For example, imagine a user supplies the paths `./tests/a` and `./tests/a/b`. The tests inside `./tests/a/b` will be collected twice.

There's a few different ways I can think of getting around this:
1. Before entering a subdirectory to look for tests, we check if it that directory has already been supplied separately by the user, and if it has we skip over it, knowing it will be dealt with. There's probably some other edge cases in here I haven't thought of though 😄 
2. We just collect all the tests from every path, then filter out the duplicates afterwards.

Option 1 is my favourite, but let me know what you think.

File: ward/run.py [hunk 2]
```diff
@@ -19,7 +19,8 @@
 
 
 @click.command()
-@click.option("-p", "--path", default=".", type=click.Path(exists=True), help="Path to tests.")
+@click.option("-p", "--path", nargs=0, default=".", type=click.Path(exists=True), help="Path to tests.")
```
Comment (darrenburns) [inline ward/run.py]: I missed this sentence in the click docs: "*For options, only a fixed number of arguments is supported. This can be configured by the nargs parameter.*" :(

I think the best option then would be to get rid of the `click.argument` below, and update this line as follows:

```suggestion
@click.option("-p", "--path", default=".", type=click.Path(exists=True), help="Path to tests.", multiple=True)
```

It means to pass in multiple paths you'll have to do e.g.

```
ward --path ./path1 \
     --path ./path2
```

This approach lets us maintain the default `.` path, meaning you can run all tests in the current directory just using `ward`, and it fixes the issue of being able to call without supplying a path.

File: ward/collect.py [hunk 3]
```diff
@@ -13,18 +13,22 @@ def is_test_module(module: pkgutil.ModuleInfo) -> bool:
     return module.name.startswith("test_")
 
 
-def get_info_for_modules(path: str) -> Generator[pkgutil.ModuleInfo, None, None]:
-    # Check for modules at the root of the specified path
-    for module in pkgutil.iter_modules([path]):
+def get_info_for_modules(path: List[str]) -> Generator[pkgutil.ModuleInfo, None, None]:
```
Comment (darrenburns) [inline ward/collect.py]: Small nitpick: you could update the param name from `path` to `paths` since multiple paths are now supplied.

File: ward/collect.py (hunk 1 above)
Comment (AlecJ) [reply ward/collect.py]: I see what you mean. I'll try to come up with a solution a bit later today

Comment (AlecJ) [reply ward/collect.py]: I ended up just creating a list of checked directories and only added directories that hadn't been seen before. It seems straight forward enough that I can't imagine any complications.

**Comment by darrenburns** [comment]
Hey, I just got round to looking at this, so sorry for the wait.

I found an issue by running the following command inside the repo itself: `ward --path . --path tests`.

The issue is that the strings `tests` and `./tests` represent the same path, despite being different strings. So when we initially track which directories we've looked at, the list contains `['tests', '.']`. Then, when looking through the `.` directory, we come across `./tests`, and assume that we haven't seen it already because that exact string isn't in the list.

The way to account for this is to use `Path` objects instead or strings to keep track of which paths have been visited.

I created a branch called `pr/41` with some changes in it that implement this (it also fixes the merge conflict). Feel free to merge that branch into this PR if that all makes sense, and let me know if you have any questions, and thanks again for this work!

P.S. Given this project is in it's early stages and didn't have any other contributors until you made this PR, I don't yet have a file listing contributors, but that should be coming this week. I'll be sure to add you when it does :)

**Comment by darrenburns** [comment]
I've merged this in and released 0.23.0b0, thanks so much for your help!
